lib/xandra/connection.ex

```diff
@@ -32,17 +38,19 @@ defmodule Xandra.Connection do
 
   def handle_prepare(%Prepared{} = prepared, options, %__MODULE__{socket: socket} = state) do
     force? = Keyword.get(options, :force, false)
+    compressor = assert_valid_compressor(state.compressor, options[:compressor])
```

Comment:
I think we should move state fields extraction to body, same as in the `handle_execute`.